docs/proposals/zeroconf.md

```diff
@@ -0,0 +1,151 @@
+# [Zeroconf](https://en.wikipedia.org/wiki/Zero-configuration_networking) Protocol Implementation
+
+## Goal
+
+Agent implements [Zero-configuration networking](https://en.wikipedia.org/wiki/Zero-configuration_networking) (hence 'Zeroconf'), a set of technologies that help discover devices and services using DNS-based discovery. There are 2 main elements: Multicast DNS (mDNS) and DNS-based Service Discovery (DNS-SD).
+
+While Zeroconf is often used in home networks (that don't often include regular DNS), Zeroconf is broadly applicable and is useful in IoT deployments in which devices are transient, there are many devices, and developers wish to dynamically manage services on these devices.
+
+These technologies require additional packages and shared libraries. Supporting Zeroconf as an Akri protocol provides a mechanism by which (Kubernetes) application developers can leverage Zeroconf technologies without having to install or be familiar with Zeroconf dependencies. The Akri protocol enables exposing these services to Kubernetes clusters enabling Kubernetes applications to utilize them.
+
+## Why Zeroconf?
+
+Zeroconf is a useful mechanism to publish services that have not only names (e.g. `device-123456`) but simple metadata (e.g. `_elevators._udp`) and limited textual data for e.g. labels. This permits scenarios where the Akri Zeroconf protocol could be configured to access e.g. a building's network and, using discovery find services representing e.g. its elevators and create Broker instances for each of the elevators. Akri would provide configuration information for each elevator to each Broker instance. Each Broker would encapsulate the (proprietary protocol and) functionality needed to interact with an elevator and could expose elevator functionality as REST-based or gRPC-based (or some other RPC mechanism) services to Kubernetes applications.
```

Comment:
In our docs, we've been keeping Broker pods separate from Akri Instances (custom resources), since every user may not choose to have Akri automatically deploy a specified broker Pod. Can we say here "and create Akri Instances for each of the elevators. Akri would provide information about each elevator to each Akri Instance and this information could then be passed to broker Pods via environment variables if a broker image is specified in the Zeroconf Configuration. Each Broker Pod..."

Reply:
I can also make this change if you'd prefer.

Reply:
Fixed: https://github.com/deislabs/akri/pull/162/commits/a2e2aa955aaf3f1279809e02b88f66ba7f8fba79